### PR TITLE
Add separate class to handle paths for Project class

### PR DESCRIPTION
**Pull request by Fuzzbender**
Split away from original INTDEV-468 PR. 

A new class that handles resource paths for Project class. 
Encapsulates paths and simplifies Project. In the future (after few PRs) will be responsible for providing basic needs in resource creation. The new methods are not yet used.

### CODE REVIEW

**Review by korva — APPROVED** [review]
Looks good! Prettier has not been run on some of the changed files

File: tools/data-handler/src/calculate.ts [hunk 1]
```diff
@@ -63,7 +62,7 @@ export class Calculate {
   }
 
   private getResourceFolder() {
```
Comment (samutoljamo-pareto) [inline tools/data-handler/src/calculate.ts:64]: This could also be in the new class

File: tools/data-handler/src/calculate.ts [hunk 2]
```diff
@@ -415,12 +414,12 @@ export class Calculate {
     const firstCard = cards[0];
     await this.setCalculateProject(firstCard); // can throw
     const cardTreeFile = join(
-      Calculate.project.calculationFolder,
+      Calculate.project.paths.calculationFolder,
```
Comment (samutoljamo-pareto) [inline tools/data-handler/src/calculate.ts:417]: There's a lot joins in the calculate.ts file. Would it be more clear, if all these paths were provided by the new class?

**Review by samutoljamo-pareto — APPROVED** [review]
Very good change

File: tools/data-handler/src/containers/template.ts [hunk 3]
```diff
@@ -90,7 +89,7 @@ export class Template extends CardContainer {
     parentCard?: Card,
   ): Promise<Card[]> {
     const templateIDMap: mappingValue[] = [];
-    const tempDestination = join(this.project.cardRootFolder, 'temp');
+    const tempDestination = join(this.project.paths.cardRootFolder, 'temp');
```
[the remark this answers is not in the document]
Comment (Fuzzbender) [reply tools/data-handler/src/containers/template.ts:92]: Yes, it would be good addition.

File: tools/data-handler/src/calculate.ts (hunk 1 above)
Comment (Fuzzbender) [reply tools/data-handler/src/calculate.ts:64]: I'll move it there.

File: tools/data-handler/src/calculate.ts (hunk 2 above)
Comment (Fuzzbender) [reply tools/data-handler/src/calculate.ts:417]: I'll check, if we good move it there as well.

**Pull request merged**